validator/src/main.rs

```diff
@@ -0,0 +1,249 @@
+use std::{net::SocketAddr, process::exit, sync::Arc, time};
+
+use anyhow::{anyhow, bail, ensure, Context, Result};
+use clap::Parser;
+use rand::distributions::{Alphanumeric, DistString};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+    sync::Notify,
+};
+use tracing::{debug, error, info, Instrument};
+
+#[derive(Parser)]
+#[clap(version)]
+struct Args {
+    #[clap(
+        long,
+        env = "CNI_VALIDATOR_LOG_LEVEL",
+        default_value = "cni_validator=info,warn"
+    )]
+    log_level: kubert::LogFilter,
+
+    #[clap(long, env = "CNI_VALIDATOR_LOG_FORMAT", default_value = "plain")]
+    log_format: kubert::LogFormat,
+
+    #[clap(long, default_value = "0.0.0.0:4140")]
+    outbound_proxy_addr: SocketAddr,
+
+    #[clap(long)]
+    target_addr: SocketAddr,
+
+    #[clap(parse(try_from_str = parse_timeout), long, default_value = "120s")]
+    timeout: std::time::Duration,
```

Comment:
Is a 2m timeout really appropriate? What can change while we wait all that time? My hunch is that we want something closer to a **2s** timeout so that this program can fail quickly.

Reply:
2m was completely arbitrary from my side. Lowering this to 2s sounds better.